modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js

```diff
@@ -189,10 +195,12 @@ const RichText = ({
 							if (editor.mode === 'source') {
 								const value = event.data.dataValue;
 
-								const sanitizedValue = value.replace(
-									/onerror="[^"]+"/gi,
-									''
-								);
+								const sanitizedValue = value
+									.replace(ON_ERROR_REGEX, '')
+									.replace(ALERT_REGEX, '')
+									.replace(INNER_HTML_REGEX, '')
+									.replace(ON_LOAD_REGEX, '')
+									.replace(ON_MOUSE_OVER_REGEX, '');
```

Comment:
Hi @danvb112, after some investigation I realized that the infinite loop was not caused by `escapeHTML` in `frontend-js`.

The root cause was probably changing event.data.dataValue inside the `onSetData` function, triggering the function repeatedly.

I found relevant [documentation](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_best_practices.html#security) that might be useful to us.

This documentation shows when we can sanitize user input to mitigate XSS attack using the `contentPreview` function. We can test to see if we can use `escapeHTML` in `contentPreview`.